base_mixin_restrict_field_access/models/restrict_field_access_mixin.py

```diff
@@ -72,6 +72,61 @@ def read(self, fields=None, load='_classic_read'):
                         self._fields[field].null(self.env))
         return result
 
+    def read_group(self, cr, uid, domain, fields, groupby, offset=0, limit=None, context=None, orderby=False, lazy=True):
+        """
+        Remove inaccessible fields from 'fields', 'groupby' and 'orderby'.
+
+        If this removes all 'fields', return no records.
+        If this removes all 'groupby', group by first remaining field.
+        If this removes 'orderby', don't specify order.
+        """
+        sanitised_fields = [f for f in fields if self._restrict_field_access_is_field_accessible(cr, uid, [], f)]
```

Comment:
this call is tricky btw, because the idea with this mixin is that developers can restrict field access per record. As you call this on no record, this will break some of the logic. I see two ways to cope with that:

- call super first, loop through the results, use the `__domain` key per row to query the records involved, and null out the field for the row if one of the rows involved doesn't allow field access. This is the cleanest solution, but horribly slow
- invent another action `read_group`, then the user of the mixin needs to take care to take sensible action for this case. This should be reflected in the docstrings of all functions which get an action passed